### PR TITLE
Feat/custom header items slot

### PR DESCRIPTION
**Pull request by alexdudescu**
Component: Suggest 

Have a slot at the top of the dropdown list to display a list of custom elements.

Todo: 
- [x] display custom items and separate with divider
- [x] correctly manage activeIndex when option is used
- [x] update tests

### CODE REVIEW

**Review by robertandreimoldoveanu — APPROVED** [review]

File: projects/angular/components/ui-suggest/src/ui-suggest.component.ts [hunk 1]
```diff
@@ -187,6 +187,30 @@ export class UiSuggestComponent extends UiSuggestMatFormFieldDirective
...
+    @Input()
+    get headerItems() {
+        return this.loading$.value || !!this.inputControl.value.trim()
+            ? []
+            : this._headerItems;
+    }
+    set headerItems(value: ISuggestValue[] | null) {
+        if (!value || isEqual(value, this._headerItems)) { return; }
+
+        if (this.enableCustomValue) {
```
Comment (anbalase) [inline projects/angular/components/ui-suggest/src/ui-suggest.component.ts:203]: a check should be performed in the setter of enableCustomValue as well

File: projects/angular/components/ui-suggest/src/ui-suggest.component.ts [hunk 2]
```diff
@@ -1190,6 +1250,20 @@ export class UiSuggestComponent extends UiSuggestMatFormFieldDirective
...
     };
 
+    private _setAndScrollToActiveIndex(value: ISuggestValue) {
+        const hasSingleSelectValue = !!value && !this.multiple;
+
+        this._setActiveIndex(
+            hasSingleSelectValue
+                ? this.headerItems!.length
+                    ? this.headerItems!.findIndex(({ id }) => id === value.id)
+                    : this.computedItemsOffset(this._items.findIndex(({ id }) => id === value.id))
```
Comment (anbalase) [inline projects/angular/components/ui-suggest/src/ui-suggest.component.ts:1260]: What if this is called with an item from the `this._items` but also we have `headerItems` defined?
will activeIndex be set correctly?

File: projects/angular/components/ui-suggest/src/ui-suggest.component.html [hunk 3]
```diff
@@ -231,10 +231,59 @@
...
+                                    intl.translateLabel(item.text) }}</span>
+                            </div>
+                            <mat-icon *ngIf="itemExpandable"
+                                      class="expand-icon">chevron_right</mat-icon>
+                        </div>
+                    </ng-template>
+                </mat-list-item>
+            </ng-container>
+
+            <mat-divider *ngIf="headerItems!.length"></mat-divider>
```
Comment (anbalase) [inline projects/angular/components/ui-suggest/src/ui-suggest.component.html:281]: can't we avoid template duplication? I'm afraid of maintenance costs when we change anything for items

Comment (anbalase) [reply projects/angular/components/ui-suggest/src/ui-suggest.component.html:281]: Consider just changing `get renderItems()`.
Divider could be added if `index && index === headerItems!.length`
Am I missing something?

File: projects/angular/components/ui-suggest/src/ui-suggest.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexdudescu) [reply projects/angular/components/ui-suggest/src/ui-suggest.component.ts:203]: nope :D

File: projects/angular/components/ui-suggest/src/ui-suggest.component.html (hunk 3 above)
Comment (alexdudescu) [reply projects/angular/components/ui-suggest/src/ui-suggest.component.html]: There was indeed an opportunity for extracting the common template section. I did opt as well for altering renderItems in the first place, but that will mess with the rangeLoad directive and accordingly with the requests.

**Review by robertandreimoldoveanu — APPROVED** [review]

**Pull request merged**